### PR TITLE
fixed loading .bin files, filter topics and example gas amounts

### LINKED ISSUE #40: Invalid contracts' binary code at deploy stage
This is a left-side contract of bridge (mainnet_contract) deployed with the bridge in the Kovan network:
https://kovan.etherscan.io/tx/0x283e8365f60c4cc7f5d62377fa0e8fb48a29b93ba65e2622962c1074b72467ba

The transaction is marked as invalid because the invalid binary of contract is sent: if check the binary, it is the hex of the real binary of contract, that is stored in `.\bridge\src\contracts\Ethereum.bridge.json`. It seem, that double hex is somewhere in the code.

### LINKED ISSUE #41: The balance of the investor in testnet doesn't change
Steps to reproduce:

1. I've started Parity as described in docs with this `parity-spec.json`:
```
{
	"name": "BridgeExample",
	"engine": {
		"instantSeal": { "params": {} }
	},
	"params": {
		"gasLimitBoundDivisor": "0x0400",
		"accountStartNonce": "0x0",
		"maximumExtraDataSize": "0x20",
		"minGasLimit": "0x1388",
		"networkID" : "0x11"
	},
	"genesis": {
		"seal": {
			"generic": "0x0"
		},
		"difficulty": "0x20000",
		"author": "0x0000000000000000000000000000000000000000",
		"timestamp": "0x00",
		"parentHash": "0x0000000000000000000000000000000000000000000000000000000000000000",
		"extraData": "0x",
		"gasLimit": "0x5B8D80"
	},
	"accounts": {
		"0000000000000000000000000000000000000001": { "balance": "1", "builtin": { "name": "ecrecover", "pricing": { "linear": { "base": 3000, "word": 0 } } } },
		"0000000000000000000000000000000000000002": { "balance": "1", "builtin": { "name": "sha256", "pricing": { "linear": { "base": 60, "word": 12 } } } },
		"0000000000000000000000000000000000000003": { "balance": "1", "builtin": { "name": "ripemd160", "pricing": { "linear": { "base": 600, "word": 120 } } } },
		"0000000000000000000000000000000000000004": { "balance": "1", "builtin": { "name": "identity", "pricing": { "linear": { "base": 15, "word": 3 } } } },
		"0xcace118480aa444aa8fe4318d97b05be927ca127": { "balance": "1606938044258990275541962092341162602522202993782792835301376" }
	}
}
```

2. and config:

```
[parity]
chain = "examples/parity_spec.json"

[account]
unlock = ["0xcace118480aa444aa8fe4318d97b05be927ca127"]
password = ["examples/parity_password"]

[mining]
engine_signer = "0xcace118480aa444aa8fe4318d97b05be927ca127"
reseal_on_txs = "all"
usd_per_tx = "0"
reseal_min_period = 0
```

3. The bridge config is:

```
[mainnet]
account = "0xcace118480aa444aa8fe4318d97b05be927ca127"
ipc = "/Users/viktor/Library/Application Support/io.parity.ethereum/jsonrpc.ipc"
required_confirmations = 0
request_timeout = 120

[mainnet.contract]
bin = "contracts/EthereumBridge.bin"

[testnet]
account = "0xcace118480aa444aa8fe4318d97b05be927ca127"
ipc = "/Users/viktor/Library/Application Support/io.parity.ethereum/jsonrpc.ipc"
required_confirmations = 0
request_timeout = 120

[testnet.contract]
bin = "contracts/KovanBridge.bin"

[authorities]
accounts = [
	"0xcace118480aa444aa8fe4318d97b05be927ca127"
]
required_signatures = 1

[transactions]
mainnet_deploy = { gas = 4538560 }
testnet_deploy = { gas = 4538560 }
```
So, the bridge listens the same instance for "testnet" and "mainnet".

4. I've started bridge: contracts are deployed successfully, the database file with the addresses of contracts is created.
5. I sent some eth to "mainnet" contract from address A = `0xcace118480aa444aa8fe4318d97b05be927ca127`. I've checked, that balance of the contract is changed. I've checked, that Deposit Event was appeared. But `balances` method for the same address A at "testnet" contract still returns 0.

What am I doing wrong?

### PR DESCRIPTION
**Pull request by debris**
closes #39 

> 1. `examples/parity_start.sh` from the documentation doesn't start Parity, because of invalid `parity_spec.json`. The structure `{ "params": {} }` is added for `instantSeal` property of `engine`

bridge requires latest version of parity (1.8.0), which is available only on master branch. This version of parity has slightly different spec format. I've created an issue to describe that - #42

> 2. The contracts' binaries from `.\bridge\src\contracts\` don't match the binaries from that, compiled with Truffle. The binaries from the Truffle compilation folder are copied to `.\bridge\src\contracts\` folder.

I'm not compiling with truffle, but with the latest version of solidity compiler (see `./tools/solc_compile.sh`). It would be good to change that and use solidity compiler attached to truffle, cause tests are run using that compiler. But, for now, I do not want to copy'n'paste compiled code from json to the file manually. It would be nice to automated script for that - #43

fixes #40

- this is my stupid bug :/ I must have refactored loading contract from the file at some point and I broke both the code and tests which were checking that :/

closes #41

there are actually two reasons

- Gas. There is no bug here. Transaction gas was not set properly. I will describe it further in the comments.
- apparently parity does not fully conform to ethereum specification for log filters. I must have changed the way empty filters topics are serialized at some point. They are still serialized to a valid format but parity has a bug and does not accept it. I temporarily fixed it here, but this will also require upstream fix in the parity repo.

### CODE REVIEW

File: examples/config.toml [hunk 1]
```diff
@@ -17,11 +17,10 @@ bin = "contracts/KovanBridge.bin"
 [authorities]
 accounts = [
 	"0x006e27b6a72e1f34c626762f3c4761547aff1421",
-	"0x006e27b6a72e1f34c626762f3c4761547aff1421",
-	"0x006e27b6a72e1f34c626762f3c4761547aff1421"
 ]
-required_signatures = 2
+required_signatures = 1
```
Comment (debris) [inline examples/config.toml]: changed example `required_signatures` to 1, so the transactions are actually relayed by a single authority

File: examples/config.toml [hunk 2]
```diff
@@ -17,11 +17,10 @@ bin = "contracts/KovanBridge.bin"
...
 
 [transactions]
-mainnet_deploy = { gas = 500000 }
-testnet_deploy = { gas = 500000 }
+mainnet_deploy = { gas = 1000000 }
+testnet_deploy = { gas = 3000000 }
+deposit_relay = { gas = 100000 }
```
Comment (debris) [inline examples/config.toml]: you need to specify `gas` for each operation. Right now the values are arbitrary, but I've created an issue to set the defaults properly #45

**Comment by debris** [comment]
@vbaranov please check if I addressed all of your issues ;)

**Comment by vbaranov** [comment]
@debris thank you. It seems, the bridge works as expected with this pull request. And all my issues should be closed with it.

**Pull request merged**